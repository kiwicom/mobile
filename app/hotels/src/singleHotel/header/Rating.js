// @flow

import * as React from 'react';
import { Stars } from '@kiwicom/react-native-app-shared';
import Translation, {
  TranslationFragment,
} from '@kiwicom/react-native-app-translations';

type Props = {|
  stars?: ?number,
  score?: ?number,
  description?: ?string,
|};

export default function Rating({ stars, score, description }: Props) {
  const reviewDelimiter = score && description ? ' ' : '';
  const review = `${score || ''}${reviewDelimiter}${description || ''}`;
  const starsDelimiter = stars && (score || description) ? ' - ' : '';

  return (
    <TranslationFragment>
      <Stars rating={stars || 0} />
      <Translation passThrough={starsDelimiter} />
      <Translation passThrough={review} />
    </TranslationFragment>
  );
}
