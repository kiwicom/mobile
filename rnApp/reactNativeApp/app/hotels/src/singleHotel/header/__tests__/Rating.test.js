// @flow

import * as React from 'react';
import renderer from 'react-test-renderer';

import Rating from '../Rating';

describe('formatRating', () => {
  it('concats stars, review score and review description', () => {
    expect(
      renderer
        .create(<Rating stars={3} score={8.6} description="Good Enough" />)
        .toJSON(),
    ).toMatchSnapshot();
  });

  it('omits the dash when stars are missing', () => {
    expect(
      renderer
        .create(<Rating score={8.6} description="Good Enough" />)
        .toJSON(),
    ).toMatchSnapshot();
  });

  it('omits the dash when review is missing', () => {
    expect(renderer.create(<Rating stars={5} />).toJSON()).toMatchSnapshot();
  });
});
