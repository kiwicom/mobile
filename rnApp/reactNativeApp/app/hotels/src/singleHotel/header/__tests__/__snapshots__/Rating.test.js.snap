// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatRating concats stars, review score and review description 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    ★★★
  </Text>
   - 
  8.6 Good Enough
</Text>
`;

exports[`formatRating omits the dash when review is missing 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    ★★★★★
  </Text>
  
  
</Text>
`;

exports[`formatRating omits the dash when stars are missing 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    
  </Text>
  
  8.6 Good Enough
</Text>
`;
